### PR TITLE
feat(cyber-pilot): integrate cyber-pilot with CyberFabric SDLC

### PR DESCRIPTION
**Pull request by ainetx**
Add Cyber Pilot adapter for Hyperspot with cf-sdlc kit supporting PRD → DESIGN → DECOMPOSITION → FEATURE → CODE workflow.

Changes:
- Add .cypilot-adapter with AGENTS.md, artifacts.json, cf-sdlc weaver
- Add IDE integrations (Claude, Cursor, Windsurf, GitHub Copilot)
- Enhance docs/checklists with expert review domains (ARCH, SEC, PERF, etc.)
- Update docs/spec-templates with Cypilot ID format (spd-*)
- Migrate todo-app examples from fdd-* to spd-* ID prefix
- Add cypilot-agents.json and .cypilot-config.json
- Add navigation rules for NEW_MODULE.md and CONTRIBUTING.md

## Summary by CodeRabbit

* **New Features**
  * Cypilot integration: artifact validation now runs in pre-release/CI and multi-agent workflows for generate/analyze/adapter.

* **Documentation**
  * Large expansion of templates, checklists, examples, ADRs and kit/rules; standardized public identifier conventions and enhanced traceability guidance.

* **Chores**
  * Added adapter/kit scaffolding, agent skill/prompt/workflow stubs, manifests/configs, submodule entry, .gitignore update, and Makefile/CI validate-artifacts target.

### CODE REVIEW

File: .spaider-adapter/weavers/cf-sdlc/artifacts/ADR/rules.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ALWAYS open and follow `{spaider_adapter_path}/specs/build-deploy.md` and `{spaider_adapter_path}/specs/compliance.md` WHEN decisions impact CI, supply-chain, or policy tooling
+
+## Hyperspot Deltas vs Original Spaider SDLC
+
+- Prefer decisions that align with repository constraints (ModKit patterns, secure ORM, OData `$select`, CI safety tooling).
+
+## Generation Checklist
+
+- [ ] Capture the problem statement, drivers, options considered, and the decision with consequences.
+- [ ] Keep ADRs immutable once ACCEPTED; supersede via a new ADR.
```
Comment (Artifizer) [inline .spaider-adapter/weavers/cf-sdlc/artifacts/ADR/rules.md]: I'd say something like: "Keep ADRs decisions immutable once ACCEPTED, allow only structural or syntax or grammar changes"

File: .spaider-adapter/weavers/cf-sdlc/artifacts/FEATURE/rules.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+- [ ] Reference the DECOMPOSITION feature ID via `id-ref:feature` (do not redefine the same feature ID).
+- [ ] Define flows/algorithms/states/requirements at implementable detail level (inputs/outputs, errors, edge cases).
+- [ ] Keep details consistent with ModKit patterns, secure ORM, and OData/OpenAPI conventions.
+- [ ] Write acceptance criteria that are testable and map to `make test`/integration/E2E expectations.
+
+## Validation Checklist
+
+- [ ] `python3 {spaider_path}/skills/spaider/scripts/spaider.py validate --artifact <path>`
+- [ ] `python3 {spaider_path}/skills/spaider/scripts/spaider.py list-ids --artifact <path>` and confirm no duplicates.
```
Comment (Artifizer) [inline .spaider-adapter/weavers/cf-sdlc/artifacts/FEATURE/rules.md]: can these 2 CLI commands be merged to simplify tooling?

File: .spaider-adapter/weavers/cf-sdlc/artifacts/PRD/rules.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+ALWAYS open and follow `../../rules.md`
+
+ALWAYS open and follow `docs/spec-templates/PRD.md` as a template
+
+ALWAYS open and follow `docs/checklists/PRD.md` as a quality checklist
+
+ALWAYS open and follow `{spaider_adapter_path}/specs/project-structure.md` WHEN choosing locations for new artifacts
+
+ALWAYS open and follow `{spaider_adapter_path}/specs/tech-stack.md` WHEN stating platform constraints or runtime assumptions
```
Comment (Artifizer) [inline .spaider-adapter/weavers/cf-sdlc/artifacts/PRD/rules.md]: Why PRD needs to know about tech stack?

File: .spaider-adapter/weavers/cf-sdlc/artifacts/FEATURE/rules.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ALWAYS open and follow `{spaider_adapter_path}/specs/security.md` and `{spaider_adapter_path}/specs/data-governance.md` WHEN specifying persistence and access control
+
+ALWAYS open and follow `{spaider_adapter_path}/specs/testing.md` WHEN writing acceptance criteria and test strategy
+
+ALWAYS open and follow `{spaider_adapter_path}/specs/performance.md` and `{spaider_adapter_path}/specs/reliability.md` WHEN specifying latency/retry/timeout behavior
+
+## Hyperspot Deltas vs Original Spaider SDLC
+
+- Original SDLC uses `SPEC` artifacts; Hyperspot uses `FEATURE` artifacts as the implementable unit.
+- The feature ID is defined in DECOMPOSITION; FEATURE references it via `id-ref:feature` (do not redefine).
```
[the remark this answers is not in the document]
Comment (ainetx) [reply .spaider-adapter/weavers/cf-sdlc/artifacts/FEATURE/rules.md]: fixed

File: .spaider-adapter/weavers/cf-sdlc/codebase/rules.md [hunk 5]
```diff
@@ -0,0 +1,555 @@
...
+```
+
+**ID reference** examples:
+
+```text
+`spd-...`
+[x] `p1` - `spd-...`
+```
+
+Any inline `` `spd-...` `` in text is treated as an ID reference.
```
[the remark this answers is not in the document]
Comment (ainetx) [reply .spaider-adapter/weavers/cf-sdlc/codebase/rules.md]: fixed

File: .cypilot-adapter/kits/cf-sdlc/artifacts/FEATURE/rules.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+## Constraints Alignment (REQUIRED)
+
+FEATURE MUST define at least one `dod` ID.
+
+- `dod` requires task+priority → use checkbox ID definition form (`- [ ] `pN` - **ID**: ...`).
+- `dod` is `to_code: true` (intended to drive implementation + traceability).
+
+Optional ID kinds allowed in FEATURE:
+
+- `flow`
```
Comment (Artifizer) [inline .cypilot-adapter/kits/cf-sdlc/artifacts/FEATURE/rules.md]: do we still need flow and algo?
and featurecontext?

File: .cypilot-adapter/kits/cf-sdlc/artifacts/FEATURE/rules.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+## CyberFabric Deltas vs Original Cypilot SDLC
+
+- CyberFabric uses `FEATURE` artifacts as the implementable unit.
+
+## Constraints Alignment (REQUIRED)
+
+FEATURE MUST define at least one `dod` ID.
+
+- `dod` requires task+priority → use checkbox ID definition form (`- [ ] `pN` - **ID**: ...`).
+- `dod` is `to_code: true` (intended to drive implementation + traceability).
```
[the remark this answers is not in the document]
Comment (ainetx) [reply .cypilot-adapter/kits/cf-sdlc/artifacts/FEATURE/rules.md]: from `constraints.json`

File: docs/checklists/CODING.md [hunk 8]
```diff
@@ -0,0 +1,678 @@
+---
```
[the remark this answers is not in the document]
Comment (ainetx) [reply docs/checklists/CODING.md]: I will think about the optimization later

**Review by Artifizer — APPROVED** [review]
